### DISCUSSION TITLE
could not boot exec format error

### DISCUSSION THREAD

**Discussion by emoxam** [ipxe/ipxe#981] [General]
Hello!
I used this part

```
:debian-installer1
cpuid --ext 29 && set arch amd64 || set arch i386
initrd http://debian.mirrors.ovh.net/debian/dists/stable/main/installer-${arch}/current/images/netboot/mini.iso
chain ${boot-url}/memdisk iso raw || goto failed
```

But i got **could not boot exec format error**

 I found 
`If you are running a UEFI version of iPXE then only EFI binaries can be loaded. PXE binaries such as pxelinux.0 and memdisk use legacy BIOS calls and can never work on UEFI.`
At https://ipxe.org/err/2e0080
And i do use ipxe-x86_64.efi at my tiny pxe server, and a string 
**chain ${boot-url}/memdisk iso raw || goto failed**
already was in the config. What should i change?

Thanks

**Comment by NiKiZe** [comment]
If you use initrd memdisk in your script, you will not see EFI flag.
Memdisk is a pcbios binary that you can't use under EFI.

Feel free to search for iso in the issues and discussions to find sevaral explanations on why what you are doing is a bad idea.

**Reply by emoxam** [reply]
This is not me doing this, these strings exists in your default config. Why are you blaming me ?

**Reply by emoxam** [reply]
String 343 and so on of file menu.ipxe

```
:debian-installer
cpuid --ext 29 && set arch amd64 || set arch i386
#sanboot --no-describe --keep ${boot-url}/debian-installer/i386/mini.iso || goto failed
#or
initrd http://debian.mirrors.ovh.net/debian/dists/stable/main/installer-${arch}/current/images/netboot/mini.iso
chain ${boot-url}/memdisk iso raw || goto failed
#or
#kernel http://debian.mirrors.ovh.net/debian/dists/stable/main/installer-${arch}/current/images/netboot/debian-installer/${arch}/linux
#initrd http://debian.mirrors.ovh.net/debian/dists/stable/main/installer-${arch}/current/images/netboot/debian-installer/${arch}/initrd.gz

```

Shoul it work out of the box ?

**Reply by NiKiZe** [reply]
What you are calling config is a script.
iPXE does not have any "default scripts"
memdisk is a pcbios binary. You can't use it in EFI mode.

Verify by using kernel valid_efi binary followed by imgstat, and you will see that it has [EFI] flag.
Do the same with memdisk. It will not have [EFI] .
This is because it is not a valid efi binary.

Where does your menu.ipxe script come from?
I would not expect memdisk to work correctly on anything other than DOS related software.

**Reply by emoxam** [reply]
I download windows package from here  http://reboot.pro/files/file/303-tiny-pxe-server/
googled by tiny px server http://www.mistyprojects.co.uk/documents/TinyPXEServer/files/pxesrv_about.htm

So how to boot debian (for example) ? Do your application goes with no examples out of the box ?

**Reply by NiKiZe** [reply]
iPXE is a network boot loader, a tool, it is used together with several other required components to get a working solution.
Just like Grub, or the Linux kernel itself.

Extract the kernel, and initrd from ISO, boot that, then check their documentation for net-boot.

The provided sample is 'chain http://boot.ipxe.org/demo/'

The sites you mention might have different samples, another site with samples is netboot.xyz.
